Utilities/DataDistribution/Common/include/Common/ReadoutDataModel.h

```diff
@@ -0,0 +1,106 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef ALICEO2_READOUT_DATAMODEL_H_
+#define ALICEO2_READOUT_DATAMODEL_H_
+
+#include "Common/Utilities.h"
+#include "Common/DataModelUtils.h"
+
+#include "O2Device/O2Device.h"
+#include "Headers/DataHeader.h"
+
+#include <vector>
+#include <map>
+#include <stdexcept>
+
+namespace o2 {
+namespace DataDistribution {
+
+using O2DataHeader = o2::Header::DataHeader;
+using namespace o2::Base;
+using namespace o2::Header;
+
+////////////////////////////////////////////////////////////////////////////////
+/// Readout - STF Builder channel multiplexer
+////////////////////////////////////////////////////////////////////////////////
+
+enum ReadoutStfBuilderObjectType {
+  eStfStart,   // info in the object itself
+  eReadoutData // an O2SubTimeFrameLinkData follows
+};
+
+struct ReadoutStfBuilderObjectInfo : public O2DataHeader {
+  ReadoutStfBuilderObjectInfo() : O2DataHeader()
+  {
+    headerSize = sizeof(ReadoutStfBuilderObjectInfo);
+    dataDescription = gDataDescriptionInfo;
+    payloadSerializationMethod = gSerializationMethodNone;
+  }
+
+  ReadoutStfBuilderObjectInfo(O2Device& pDevice, const std::string& pChan, const int pChanId)
+  {
+    if (!receive(pDevice, pChan, pChanId))
+      throw std::runtime_error("receive error");
+  }
+
+  bool send(O2Device& pDevice, const std::string& pChan, const int pChanId);
+  bool receive(O2Device& pDevice, const std::string& pChan, const int pChanId);
+
+  // multiplex on this
+  ReadoutStfBuilderObjectType mObjectType;
+
+  // might as well piggyback this info to avoid dedicated message
+  std::uint64_t mStfId;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+/// CRU Link
+////////////////////////////////////////////////////////////////////////////////
+
+// This data struct is transmitted 'atomically' by the Readout
+struct O2CruLinkHeader : public O2DataHeader {
```

Comment:
`O2CRULinkHeader` must inherit from o2::Header::BaseHeader`; regarding naming, the O2 prefix should be removed as we use namespace o2